state/redis/redis.go

```diff
@@ -25,20 +24,69 @@ import (
 	"github.com/go-redis/redis/v8"
 	jsoniter "github.com/json-iterator/go"
 
+	"github.com/dapr/components-contrib/contenttype"
 	rediscomponent "github.com/dapr/components-contrib/internal/component/redis"
+	daprmetadata "github.com/dapr/components-contrib/metadata"
 	"github.com/dapr/components-contrib/state"
+	"github.com/dapr/components-contrib/state/query"
 	"github.com/dapr/components-contrib/state/utils"
 	"github.com/dapr/kit/logger"
 )
 
 const (
-	setQuery                 = "local var1 = redis.pcall(\"HGET\", KEYS[1], \"version\"); if type(var1) == \"table\" then redis.call(\"DEL\", KEYS[1]); end; local var2 = redis.pcall(\"HGET\", KEYS[1], \"first-write\"); if not var1 or type(var1)==\"table\" or var1 == \"\" or var1 == ARGV[1] or (not var2 and ARGV[1] == \"0\") then redis.call(\"HSET\", KEYS[1], \"data\", ARGV[2]); if ARGV[3] == \"0\" then redis.call(\"HSET\", KEYS[1], \"first-write\", 0); end; return redis.call(\"HINCRBY\", KEYS[1], \"version\", 1) else return error(\"failed to set key \" .. KEYS[1]) end"
-	delQuery                 = "local var1 = redis.pcall(\"HGET\", KEYS[1], \"version\"); if not var1 or type(var1)==\"table\" or var1 == ARGV[1] or var1 == \"\" or ARGV[1] == \"0\" then return redis.call(\"DEL\", KEYS[1]) else return error(\"failed to delete \" .. KEYS[1]) end"
+	setDefaultQuery = `
+	local etag = redis.pcall("HGET", KEYS[1], "version");
+	if type(etag) == "table" then
+	  redis.call("DEL", KEYS[1]);
+	end;
+	local fwr = redis.pcall("HGET", KEYS[1], "first-write");
+	if not etag or type(etag)=="table" or etag == "" or etag == ARGV[1] or (not fwr and ARGV[1] == "0") then
+	  redis.call("HSET", KEYS[1], "data", ARGV[2]);
+	  if ARGV[3] == "0" then
+	    redis.call("HSET", KEYS[1], "first-write", 0);
+	  end;
+	  return redis.call("HINCRBY", KEYS[1], "version", 1)
+	else
+	  return error("failed to set key " .. KEYS[1])
+	end`
+	delDefaultQuery = `
+	local etag = redis.pcall("HGET", KEYS[1], "version");
+	if not etag or type(etag)=="table" or etag == ARGV[1] or etag == "" or ARGV[1] == "0" then
+	  return redis.call("DEL", KEYS[1])
+	else
+	  return error("failed to delete " .. KEYS[1])
+	end`
+	setJSONQuery = `
+	local etag = redis.pcall("JSON.GET", KEYS[1], ".version");
+	if type(etag) == "table" then
+	  redis.call("JSON.DEL", KEYS[1]);
+	end;
+	if not etag or type(etag)=="table" or etag == "" then
+	  etag = ARGV[1];
+	end;
+	local fwr = redis.pcall("JSON.GET", KEYS[1], ".first-write");
+	if etag == ARGV[1] or ((not fwr or type(fwr) == "table") and ARGV[1] == "0") then
```

Comment:
Why is there a check of `type(fwr) == "table"` here but not in line 43 in the `setDefaultQuery`?